src/array.jl

```diff
@@ -212,6 +212,31 @@ end
 
             convert($A{T, N, R}, A)
         end
+
+        function Base.vcat{T,N,R}(A1::$A{T, N, R}, An::$A...)
+            As = (A1, An...)
+            levels = unique(T[[a.pool.levels for a in As]...;])
```

Comment:
`T` for the first input shouldn't win over the other inputs. Use `Base.promote_eltype` to compute the best element type.